### DISCUSSION TITLE
add support for darwin hosted aarch64

### DISCUSSION THREAD

**Discussion by hkiel** [aros-development-team/AROS#628] [Ideas]
Is there a plan to support AROS hosted on recent macs with M1 (aarch64)?

**Comment by Kalamatee** [comment]
If someone wants to put the work in to make it happen ;D

The main issue would be getting a nightly build done - we rely on the azure pipelines, so that would need a suitable mac environment to build under or be able to get the needed components to cross compile (which is notoriously difficult without an apple ID).

**Reply by bitplane** [reply]
I got Codex to start with PureDarwin's build system and grind away at it inside a container until it could build Darwin itself plus my 9p server, and published the toolchain [here](https://github.com/bitplane/mountin/pkgs/container/mountin%2Fbuilder%2Fcompiler%2Fpuredarwin%2F17.4%2Fx86_64). It's only for Intel Macs, and doesn't have any of the SDK stuff that you'd need to integrate with an actual desktop, and any build made with it would be unsigned and untrusted. But it should be possible to start from there and make an X11 build for people who have an X server. I don't have a Mac at the moment myself, but if someone really wanted to do it I think that image and [its source](https://github.com/bitplane/PureDarwin/tree/linux-build) might be a fun starting point - it should be possible to apply the same changes to a later aarch64 Darwin release and get *something* built anyway
